From 274b56503a90c266707d3e61e2881fdf3dd7c25b Mon Sep 17 00:00:00 2001
From: Rajesh Balamohan <rbalamohan at apache dot org>
Date: Fri, 16 Sep 2016 00:28:16 -0700
Subject: [PATCH 0941/1164] CDH-49262: HIVE-14764 : Enabling
 "hive.metastore.metrics.enabled" throws OOM in
 HiveMetastore (Rajesh Balamohan, reviewed by
 Vaibhav Gumashta)

(cherry picked from commit 7dd1d3694856a2fd55eba1355e81e5fa71477b0d)

Change-Id: Ie0e75c33f7ea1dcaf8445b8e43367a88d9babd0c
---
 .../hadoop/hive/metastore/HiveMetaStore.java       |   50 +++++++++++---------
 1 file changed, 27 insertions(+), 23 deletions(-)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
index 80e2bde..c4238d1 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
@@ -4477,7 +4477,7 @@ public ColumnStatistics get_table_column_statistics(String dbName, String tableN
       dbName = dbName.toLowerCase();
       tableName = tableName.toLowerCase();
       colName = colName.toLowerCase();
-      startFunction("get_column_statistics_by_table: db=" + dbName + " table=" + tableName +
+      startFunction("get_column_statistics_by_table", ": db=" + dbName + " table=" + tableName +
                     " column=" + colName);
       ColumnStatistics statsObj = null;
       try {
@@ -4486,7 +4486,7 @@ public ColumnStatistics get_table_column_statistics(String dbName, String tableN
         assert statsObj.getStatsObjSize() <= 1;
         return statsObj;
       } finally {
-        endFunction("get_column_statistics_by_table: ", statsObj != null, null, tableName);
+        endFunction("get_column_statistics_by_table", statsObj != null, null, tableName);
       }
     }
 
@@ -4495,7 +4495,7 @@ public TableStatsResult get_table_statistics_req(TableStatsRequest request)
         throws MetaException, NoSuchObjectException, TException {
       String dbName = request.getDbName().toLowerCase();
       String tblName = request.getTblName().toLowerCase();
-      startFunction("get_table_statistics_req: db=" + dbName + " table=" + tblName);
+      startFunction("get_table_statistics_req", ": db=" + dbName + " table=" + tblName);
       TableStatsResult result = null;
       List<String> lowerCaseColNames = new ArrayList<String>(request.getColNames().size());
       for (String colName : request.getColNames()) {
@@ -4506,7 +4506,7 @@ public TableStatsResult get_table_statistics_req(TableStatsRequest request)
         result = new TableStatsResult(
             cs == null ? Lists.<ColumnStatisticsObj>newArrayList() : cs.getStatsObj());
       } finally {
-        endFunction("get_table_statistics_req: ", result == null, null, tblName);
+        endFunction("get_table_statistics_req", result == null, null, tblName);
       }
       return result;
     }
@@ -4519,8 +4519,9 @@ public ColumnStatistics get_partition_column_statistics(String dbName, String ta
       tableName = tableName.toLowerCase();
       colName = colName.toLowerCase();
       String convertedPartName = lowerCaseConvertPartName(partName);
-      startFunction("get_column_statistics_by_partition: db=" + dbName + " table=" + tableName +
-          " partition=" + convertedPartName + " column=" + colName);
+      startFunction("get_column_statistics_by_partition",
+              ": db=" + dbName + " table=" + tableName
+              + " partition=" + convertedPartName + " column=" + colName);
       ColumnStatistics statsObj = null;
 
       try {
@@ -4532,7 +4533,7 @@ public ColumnStatistics get_partition_column_statistics(String dbName, String ta
         }
         statsObj = list.get(0);
       } finally {
-        endFunction("get_column_statistics_by_partition: ", statsObj != null, null, tableName);
+        endFunction("get_column_statistics_by_partition", statsObj != null, null, tableName);
       }
       return statsObj;
     }
@@ -4542,7 +4543,7 @@ public PartitionsStatsResult get_partitions_statistics_req(PartitionsStatsReques
         throws MetaException, NoSuchObjectException, TException {
       String dbName = request.getDbName().toLowerCase();
       String tblName = request.getTblName().toLowerCase();
-      startFunction("get_partitions_statistics_req: db=" + dbName + " table=" + tblName);
+      startFunction("get_partitions_statistics_req", ": db=" + dbName + " table=" + tblName);
 
       PartitionsStatsResult result = null;
       List<String> lowerCaseColNames = new ArrayList<String>(request.getColNames().size());
@@ -4563,7 +4564,7 @@ public PartitionsStatsResult get_partitions_statistics_req(PartitionsStatsReques
         }
         result = new PartitionsStatsResult(map);
       } finally {
-        endFunction("get_partitions_statistics_req: ", result == null, null, tblName);
+        endFunction("get_partitions_statistics_req", result == null, null, tblName);
       }
       return result;
     }
@@ -4587,12 +4588,12 @@ public boolean update_table_column_statistics(ColumnStatistics colStats)
 
       List<ColumnStatisticsObj> statsObjs =  colStats.getStatsObj();
 
+      startFunction("write_column_statistics", ":  db=" + dbName
+          + " table=" + tableName + " column=" + colName);
       for (ColumnStatisticsObj statsObj:statsObjs) {
         colName = statsObj.getColName().toLowerCase();
         statsObj.setColName(colName);
         statsObj.setColType(statsObj.getColType().toLowerCase());
-        startFunction("write_column_statistics:  db=" + dbName + " table=" + tableName +
-          " column=" + colName);
       }
 
      colStats.setStatsDesc(statsDesc);
@@ -4604,7 +4605,7 @@ public boolean update_table_column_statistics(ColumnStatistics colStats)
         ret = getMS().updateTableColumnStatistics(colStats);
         return ret;
       } finally {
-        endFunction("write_column_statistics: ", ret != false, null, tableName);
+        endFunction("write_column_statistics", ret != false, null, tableName);
       }
     }
 
@@ -4633,12 +4634,13 @@ public boolean update_partition_column_statistics(ColumnStatistics colStats)
 
       List<ColumnStatisticsObj> statsObjs =  colStats.getStatsObj();
 
+      startFunction("write_partition_column_statistics",
+          ":  db=" + dbName + " table=" + tableName
+              + " part=" + partName + "column=" + colName);
       for (ColumnStatisticsObj statsObj:statsObjs) {
         colName = statsObj.getColName().toLowerCase();
         statsObj.setColName(colName);
         statsObj.setColType(statsObj.getColType().toLowerCase());
-        startFunction("write_partition_column_statistics:  db=" + dbName + " table=" + tableName +
-          " part=" + partName + "column=" + colName);
       }
 
       colStats.setStatsDesc(statsDesc);
@@ -4652,7 +4654,7 @@ public boolean update_partition_column_statistics(ColumnStatistics colStats)
         ret = getMS().updatePartitionColumnStatistics(colStats, partVals);
         return ret;
       } finally {
-        endFunction("write_partition_column_statistics: ", ret != false, null, tableName);
+        endFunction("write_partition_column_statistics", ret != false, null, tableName);
       }
     }
 
@@ -4667,8 +4669,9 @@ public boolean delete_partition_column_statistics(String dbName, String tableNam
         colName = colName.toLowerCase();
       }
       String convertedPartName = lowerCaseConvertPartName(partName);
-      startFunction("delete_column_statistics_by_partition: db=" + dbName + " table=" + tableName +
-                    " partition=" + convertedPartName + " column=" + colName);
+      startFunction("delete_column_statistics_by_partition",": db=" + dbName
+          + " table=" + tableName + " partition=" + convertedPartName
+          + " column=" + colName);
       boolean ret = false;
 
       try {
@@ -4676,7 +4679,7 @@ public boolean delete_partition_column_statistics(String dbName, String tableNam
         ret = getMS().deletePartitionColumnStatistics(dbName, tableName,
                                                       convertedPartName, partVals, colName);
       } finally {
-        endFunction("delete_column_statistics_by_partition: ", ret != false, null, tableName);
+        endFunction("delete_column_statistics_by_partition", ret != false, null, tableName);
       }
       return ret;
     }
@@ -4692,14 +4695,14 @@ public boolean delete_table_column_statistics(String dbName, String tableName, S
       if (colName != null) {
         colName = colName.toLowerCase();
       }
-      startFunction("delete_column_statistics_by_table: db=" + dbName + " table=" + tableName +
-                    " column=" + colName);
+      startFunction("delete_column_statistics_by_table", ": db=" + dbName
+          + " table=" + tableName + " column=" + colName);
 
       boolean ret = false;
       try {
         ret = getMS().deleteTableColumnStatistics(dbName, tableName, colName);
       } finally {
-        endFunction("delete_column_statistics_by_table: ", ret != false, null, tableName);
+        endFunction("delete_column_statistics_by_table", ret != false, null, tableName);
       }
       return ret;
    }
@@ -6050,7 +6053,8 @@ public AggrStats get_aggr_stats_for(PartitionsStatsRequest request)
         throws NoSuchObjectException, MetaException, TException {
       String dbName = request.getDbName().toLowerCase();
       String tblName = request.getTblName().toLowerCase();
-      startFunction("get_aggr_stats_for: db=" + request.getDbName() + " table=" + request.getTblName());
+      startFunction("get_aggr_stats_for", ": db=" + request.getDbName()
+          + " table=" + request.getTblName());
 
       List<String> lowerCaseColNames = new ArrayList<String>(request.getColNames().size());
       for (String colName : request.getColNames()) {
@@ -6067,7 +6071,7 @@ public AggrStats get_aggr_stats_for(PartitionsStatsRequest request)
             lowerCaseColNames));
         return aggrStats;
       } finally {
-          endFunction("get_partitions_statistics_req: ", aggrStats == null, null, request.getTblName());
+          endFunction("get_aggr_stats_for", aggrStats == null, null, request.getTblName());
       }
 
     }
-- 
1.7.9.5

